src/main/java/org/example/LoginPrompt.java

```diff
@@ -0,0 +1,23 @@
+package org.example;
+import javax.swing.*;
+
+public class LoginPrompt {
+
+    public static void main(String[] args) {
```

Comment:
Your project now has two "public static void main" methods. If you run the LoginPrompt main, you will execute the code in run(). 
If you then run the Main main, you will execute the code in that method (and any subsequent method calls in other classes).

But these would be two separate instances. They are both running, but they cannot speak to each other. 

You might find that if you change LoginPrompt's main method to be a normal method, e.g. `public void initialise() {...`, you will find that in Main's main method, you can create a new instance of Login Prompt and then call the initialise method and your Swing UI will be executed and displayed:

```
... 
final LoginPrompt loginPrompt = new LoginPrompt();
loginPrompt.initialise();
...

```